### DISCUSSION TITLE
Arch linux package is out of date + compiling from source doesn't work?

### DISCUSSION THREAD

**Discussion by marekor555** [tinygo-org/tinygo#4590] [Q&A] [answered]
Hi,
The [package](https://archlinux.org/packages/extra/x86_64/tinygo/) was flagged out of date on Aug. 20th and still hasn't been updated. Should I create an issue on the repo or contact the maintainer? 
Also, I tried compiling it from the source using `go install .`, but the compilation of my code always errors out like this:
```
> tinygo flash --target=xiao main.go

package blink
	imports machine
	imports device/sam: build constraints exclude all Go files in /home/marekor/.cache/tinygo/goroot-e6c948e66c27064830c38965779161c5a627da12b40d410e8c3bc43e6949a562/src/device/sam
> tinygo flash --target pico main.go
/home/marekor/tinygo/tinygo/src/machine/machine_rp2040.go:6:2: package device/rp is not in std (/home/marekor/.cache/tinygo/goroot-56cfc92cdd0a2d85f844b4a48f4b186cd7897fe783ee980d24817534792436d4/src/device/rp)
```

with `tinygo flash --target arduino main.go` the output is long, but it mostly boils down to `/home/marekor/tinygo/tinygo/src/machine/machine_atmega.go:41:6: undefined: avr.<something>`

I've tried everything, but I can't find what is wrong with the code/compilation method/command.

Output of `tinygo version`: `tinygo version 0.34.0 linux/amd64 (using go version go1.23.2 and LLVM version 18.1.8)`

Here is the code I use:
```
package main

import (
	"machine"
	"time"
)

func main() {
	led := machine.Pin(25)
	led.Configure(machine.PinConfig{Mode: machine.PinOutput})
	for {
		println("hello")
		led.High()
		time.Sleep(time.Millisecond * 250)
		led.Low()
		time.Sleep(time.Millisecond * 250)
	}
}
```

**Comment by aykevl** [comment, marked as answer]
For the Arch package, you need to contact the Arch package manager. We don't manage that one.

Also, you need to do a few more things than just `go install`. See this page: https://tinygo.org/docs/guides/build/additional-requirements/

**Reply by marekor555** [reply]
Thanks for the reply. I was sure I did that, but turns out I missed it. I've run make on both llvm-source and gen-device. That almost works, but I get this message: 
```
❯ tinygo flash --target=xiao main.go
none of these commands were found in your $PATH: ld.lld-18 ld.lld
```
though ld and ldd commands work when typed into the terminal my llvm version on the system is 18.1.8-4.

EDIT:
I've compiled tinygo with manual llvm build, which failed the same as `go install .` before, it works now, thanks!

**Reply by aykevl** [reply]
> though ld and ldd commands work when typed into the terminal my llvm version on the system is 18.1.8-4.

What about `ld.lld`, which is what TinyGo is looking for?
(Unfortunately I can't really run Arch Linux on my system for testing).

**Reply by marekor555** [reply]
when using custom llvm (manual llvm setup in wiki) it works, maybe my setup has issues? I have both ld and ldd installed

**Reply by marekor555** [reply]
I tested tinygo on clean install of archlinux with `go install .` and it still errors out the same way. Installing on ubuntu 24.04 LTS works. It's probably a missing package. 
EDIT: 
I have just realised what a stupid mistake I have made, I've red `lld` as `ldd`. After installing `lld` it works. I'm quite new to linux, sorry for that.

**Reply by aykevl** [reply]
Well I also missed that, so don't worry. I keep confusing those names.
